src/finch/codegen/numpy_buffer.py

```diff
@@ -90,7 +90,7 @@ class NumpyBufferFormat(CBufferFormat, NumbaBufferFormat):
     """
 
     def __init__(self, dtype: type):
-        self._dtype = dtype
+        self._dtype = np.dtype(dtype).type
 
     def __eq__(self, other):
         if not isinstance(other, type(self)):
```

Comment:
Maybe `type(other) is type(self)`?

Types are singletons, so comparing them with equality is inefficient; and this statement accepts subclasses with `other`

Reply:
Do the types need to be exactly the same for equality to hold? Or can `instance_of_parent == instance_of_subclass` be true as well?

If the former, we should really do `type(self) is not type(other)`.

If the latter, we should do `isinstance(other, CommonParentClass)` to make sure everything stays symmetrical. With the current code; `instance_of_subclass == instance_of_parent` will never be true.